annotated.go

```diff
@@ -312,6 +316,33 @@ func (ann *annotated) Build() (interface{}, error) {
 	return newFn.Interface(), nil
 }
 
+// checks whether the target function is either
+// returning an fx.Out struct or an taking in a
+// fx.In struct as a parameter.
+func (ann *annotated) typeCheckOrigFn() error {
+	ft := reflect.TypeOf(ann.Target)
+	for i := 0; i < ft.NumOut(); i++ {
+		ot := ft.Out(i)
+		if ot.Kind() != reflect.Struct {
+			continue
+		}
+		if ot.NumField() > 0 && ot.Field(0).Type == reflect.TypeOf(Out{}) {
+			return fmt.Errorf("fx.Out structs cannot be annotated")
+		}
```

Comment:
Could we collapse this into a single if condition?

```lang=go
if ot.Kind() == reflect.Struct && ot.NumField() > 0 && ... {
...
}
```
Although maybe it is more readable right now. I'll leave it up to you.

Reply:
yeah that's a bit messy.. i separated it out on purpose for readability. I'll leave it as is.